.github/workflows/build.yml

```diff
@@ -13,11 +13,23 @@ on:
   workflow_dispatch:
 
 jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - run: pip install pre-commit
+      - run: pre-commit --version
+      - run: pre-commit install
+      - run: pre-commit run black --all-files
+      - run: pre-commit run isort --all-files
+      - run: pre-commit run doc8 --all-files
   pretest:
+    needs: pre-commit
```

Comment:
Can we not wait for pre-commit? If we could set the execution of pre-commit in tox we could have `toxenv: [pylint, docs, pre-commit]` so that the main tests still depends on the same pretests.

Reply:
Running them through tox would make it possible to include the pre-commit tests in the current pre-test matrix. Having the pre-commit tests available in the tox config would also make it possible to run these tests locally if necessary outside of the pre-commit hook.

Reply:
Note: The tests can be run locally with pre-commit run <hook_name>, and either `--all-files`, or `--files <files>`.